python_modules/dagster/dagster/_core/definitions/metadata/source_code.py

```diff
@@ -232,27 +266,44 @@ def link_to_source_control(
             The asset definitions to which source control metadata should be attached.
             Only assets with local file code references (such as those created by
             `with_source_code_references`) will be converted.
-        source_control_url (str): The base URL for the source control system. For example,
+        git_url (str): The base URL for the source control system. For example,
             "https://github.com/dagster-io/dagster".
-        source_control_branch (str): The branch in the source control system, such as "master".
-        repository_root_absolute_path (Union[Path, str]): The absolute path to the root of the
-            repository on disk. This is used to calculate the relative path to the source file
-            from the repository root and append it to the source control URL.
+        git_branch (str): The branch in the source control system, such as "master".
+        git_file_path_mapping (FilePathMappingFn):
```

Comment:
Does this arg name need the "git" prefix?

Also we typically don't include type aliases in API docs, so would be better to just spell this out using `Callable`